Gems/Atom/Feature/Common/Code/Include/Atom/Feature/Mesh/MeshFeatureProcessorInterface.h

```diff
@@ -43,6 +44,9 @@ namespace AZ
             using MeshHandle = StableDynamicArrayHandle<ModelDataInstance>;
             using ModelChangedEvent = Event<const Data::Instance<RPI::Model>>;
 
+            //! Returns the object id for a mesh handle.
+            virtual TransformServiceFeatureProcessorInterface::ObjectId GetObjectId(const MeshHandle& meshHandle) const = 0;
```

Comment:
I'm not opposed to exposing this here, but I think we need to put some more thought into how we want to expose ObjectId's in Atom, since there are other use cases where people want to be able to associate a component with an ObjectId.

Reply:
👍